### PR TITLE
dotCMS/core#15509 Upgrade the dot-page-selector.component

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts [hunk 1]
```diff
@@ -143,22 +145,27 @@ export class DotPageSelectorService {
...
+        return obj;
     }
 
     private getSites(param: string, specific?: boolean): Observable<DotPageSelectorResults> {
         let query = '+contenttype:Host -identifier:SYSTEM_HOST +host.hostName:';
         query += specific ? this.getSiteName(param) : `*${this.getSiteName(param)}*`;
-
-        return this.coreWebService.requestView({
-                body: this.getRequestBodyQuery(query),
+        debugger;
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: remove

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.spec.ts [hunk 2]
```diff
@@ -105,6 +122,26 @@ describe('Service: DotPageSelector', () => {
         expect(this.lastConnection.request._body).toEqual(hostQuery);
     });
 
+    it('should make a host search but limit to MAX_RESULTS_SIZE if string is empty', () => {
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.spec.ts]: I think really we are not testing that the result is limit to MAX_RESULTS_SIZE

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts [hunk 3]
```diff
@@ -177,6 +177,15 @@ describe('DotPageSelectorComponent', () => {
...
 
+    it('should not search for pages if has less than 3 characters', () => {
+        component.results = Object.assign({}, mockDotSiteSelectorResults);
+        spyOn(dotPageSelectorService, 'search');
+        autocomplete.triggerEventHandler('completeMethod', invalidSearchPageObj);
+
+        expect(dotPageSelectorService.search).not.toHaveBeenCalled();
+        expect(component.results.data.length).toBe(0);
+    });
+
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts]: yes, here https://github.com/dotCMS/core-web/blob/master/src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts#L174

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts [hunk 4]
```diff
@@ -200,7 +200,7 @@ export class DotPageSelectorComponent implements ControlValueAccessor {
     }
 
     private validSearch (param: any): boolean {
-       return this.cleanInput(param).length && (param.query.startsWith('//') ? param.query.length >= 5 : true);
+       return this.cleanInput(param).length && (!param.query.startsWith('//') ? param.query.length >= 3 : true);
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts]: done here, https://github.com/dotCMS/core-web/blob/master/src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts#L180 I just changed the value of searchHostObj

**Pull request merged**